cpp/src/request_tag_multi.cpp

```diff
@@ -22,15 +20,20 @@ namespace ucxx {
 RequestTagMulti::RequestTagMulti(std::shared_ptr<Endpoint> endpoint,
                                  const ucp_tag_t tag,
                                  const bool enablePythonFuture)
-  : _endpoint(endpoint), _send(false), _tag(tag)
+  : Request(endpoint,
+            std::make_shared<DelayedSubmission>(true, nullptr, 0, 0),
+            std::string("tagMultiSend"),
+            enablePythonFuture),
+    _send(false),
+    _tag(tag)
 {
   ucxx_trace_req("RequestTagMulti::RequestTagMulti [recv]: %p, tag: %lx", this, _tag);
 
   auto worker = endpoint->getWorker();
   if (enablePythonFuture) _future = worker->getFuture();
 
   ucxx_debug("RequestTagMulti created: %p", this);
-  callback(UCS_OK);
+  recvCallback(UCS_OK);
```

Comment:
Remind me why we call the callback on construction of the request?

Reply:
Ah, is it to kick off the `recvHeader` call?

Reply:
That's right. I think this is not the best naming, but essentially `recvCallback` is a recursive method, so we kick it off here.